### PR TITLE
Merge develop into pre/2.8

### PR DESCRIPTION
**Pull request by yaugenst-flex**
Supersedes #1840

Quite a lot of conflicts, would be great to have some other pairs of eyes on this.

### CODE REVIEW

**Comment by momchil-flex** [comment]
May be better for another pair of eyes to take a look at the conflicting pieces of the code only rather than everything?

Out of curiosity, did you merge and then rebase? Did the rebase introduce more conflicts? For these types of things  (merge develop to and from release branches) we have been doing merging without rebasing.

**Comment by yaugenst-flex** [comment]
@tylerflex @momchil-flex I split the merge into fast-forwarded changes and a commit with the conflict resolutions now ([08768ea](https://github.com/flexcompute/tidy3d/pull/1846/commits/08768ea279628bfd3886b793d92fd2d35e82c034)). This is not perfect as you don't get any insight into what was conflicting and how it was resolved, but at least you can see which conflicting changes are getting merged into `pre/2.8`...

File: CHANGELOG.md [hunk 1]
```diff
@@ -1245,7 +1267,8 @@ which fields are to be projected is now determined automatically based on the me
 - Job and Batch classes for better simulation handling (eventually to fully replace webapi functions).
 - A large number of small improvements and bug fixes.
 
-[Unreleased]: https://github.com/flexcompute/tidy3d/compare/v2.7.0...develop
+[Unreleased]: https://github.com/flexcompute/tidy3d/compare/v2.7.1...develop
```
Comment (tylerflex) [inline CHANGELOG.md]: you'll want to also add a tag for 2.8.0rc1 here

**Review by tylerflex — APPROVED** [review]
Conflicts look resolved fine to me. Do all of the tests pass?

Note: changelog is missing a link for 2.8.0 but that can be cleaned up later too.

**Review by daquinteroflex — APPROVED** [review]
Nice, thanks! This will fix the sync issue. I've had a glance over the rest of the file changes only too but unsure if any specific conflicts you're concerned about too.

**Pull request merged**